source/extensions/quic_listeners/quiche/udp_gso_batch_writer.cc

```diff
@@ -0,0 +1,129 @@
+#include "extensions/quic_listeners/quiche/udp_gso_batch_writer.h"
+
+#include "common/network/io_socket_error_impl.h"
+
+#include "extensions/quic_listeners/quiche/envoy_quic_utils.h"
+
+namespace Envoy {
+namespace Quic {
+namespace {
+Api::IoCallUint64Result convertQuicWriteResult(quic::WriteResult quic_result) {
+  if (quic_result.status == quic::WRITE_STATUS_OK) {
```

Comment:
I would prefer a switch statement in case there is a new state introduced in QuicPacketWriter.

Reply:
Converted this into Switch Statements.